### PR TITLE
Clean code in parse_depends.R

### PR DESCRIPTION
**Pull request by hsonne**
Hello, 
I propose some slight cleaning of the code in parse_depends.R to which my colleague Michael proposed one new line in his pull request #200. You can follow my changes step by step by looking at my commits in the branch clean/parse_depends. My main motivation was to avoid code duplication and to separate the definition of URLs from the logic in if-else-statements. I added some tests that check the proper assignment of URLs. Michael's change was reintegrated with my commit 7e9157f. I hope that you agree with my changes and that I can help you make your code a bit cleaner. I still see much room of improvement. Are you interested in further collaboration?
Kind regards,
Hauke

### CODE REVIEW

**Comment by maelle** [comment]
Cool, thanks!

Could you please solve the merge conflicts (don't reintegrate the change by your colleague) and make sure the tests pass? 🙏

File: R/parse_depends.R [hunk 1]
```diff
@@ -16,83 +20,119 @@ format_depend <- function(package, version, remote_provider){
...
 # helper to get system dependencies
-get_sys_links <- function(pkg, description = ""){
-  out1 <- jsonlite::fromJSON(sprintf("https://sysreqs.r-hub.io/pkg/%s", pkg), simplifyVector = FALSE)
-  out2 <- jsonlite::fromJSON(sprintf("https://sysreqs.r-hub.io/map/%s", curl::curl_escape(description)), simplifyVector = FALSE)
-  sysreqs <- unique(c(sapply(out1, names), sapply(out2, names)))
-  sprintf("https://sysreqs.r-hub.io/get/%s", sysreqs)
+get_sys_links <- function(pkg, description = "") {
+
+  # Define helper functions
+  to_url <- function(a, b) sprintf("https://sysreqs.r-hub.io/%s/%s", a, b)
```
Comment (maelle) [inline R/parse_depends.R]: can you define the two helpers functions (`to_url()`, `get_json_names()` outside of this function and preferentially with `{}`?

Comment (hsonne) [reply R/parse_depends.R]: Good idea. I renamed the functions to make clear that they relate to "r-hub":

* `to_url()` -> `to_rhub_url()`
* `get_json_names()` -> `get_rhub_json_names()`

Comment (maelle) [reply R/parse_depends.R]: Awesome. Now let's just wait for Travis CI to run (lot of builds in the ropensci account at the moment, I think)!

**Comment by maelle** [comment]
I think the failure of one of Travis builds is due to the absence of `GITHUB_PAT` so will merge. Thank you! Don't hesitate to open issues to propose new PRs going forward. 😸

**Pull request merged**